Update: Add Donors / Embed Mailing List / Improve Responsiveness

julientregoat opened the pull request:
Here is the next update to the webpage. 

![Update Preview](https://i.imgur.com/fxDWt46.jpg)

I pushed back my focus on integrating Medium / a feed page until my next update.

In this update:

- 'Join Our Mailing List' now toggles an email input to subscribe without leaving the page
  - I removed the link to the mailing list from the top as a result. Didn't think it made sense to embed it and also have a link.
- Donation section added to main page
  - Added our list of donors
  - Added donation info, address, and link to contract on Etherscan
- Improved website responsiveness
  - Fixed margins and other things that were affecting mobile views
  - Adjusted site width to remove unnecessary horizontal scroll bar

prestonvanloon commented on src/App.css:
```diff
@@ -2,10 +2,11 @@
 body {
   height: 100%;
-  width: 100%;
+  width: 99%;
```
Why?

julientregoat replied:
There was a constant, unnecessary horizontal scroll bar that was the result of extra space. It also threw off the page symmetry on page load. I figured the only thing that could be asking for this much of the page was the body, so I tried this and it worked

prestonvanloon commented on src/about/intro/Intro.js:
```diff
@@ -0,0 +1,75 @@
+      </div>
+
+      { formVisibility ?
+        <MailchimpSubscribe
```
Can this be abstracted to another component such that you can just write something like

```
<MailchimpSubscribeButton />
```

and the rendering, logic, and everything else is contained in that component?

julientregoat replied:
on it!

prestonvanloon commented on src/about/donations/DonationsContainer.js:
```diff
@@ -0,0 +1,25 @@
+        </a>
+        <p>Thank you to all who have contributed to Prysmatic Labs; we are eternally grateful. Check 
+        out some of our awesome patrons below.</p>
+        <Donors/>
```
@rauljordan what do you think about the nomenclature here?

ECF and Ethereum Foundation are granters, i.e. the money given to us in the form of a grant has some stipulations and is not the same as a donation in the sense this is something we applied for and that we have to give the grant money back if we misuse it. 

Maybe we could split this up to have a Grant Awards section with link to the specific award annoucements and then also have a Donations section with links to the donator's web address and  donation transaction hash(s).

prestonvanloon replied:
@julientregoat Let's identify donors vs granters. It can be either by splitting these up into two sections or renaming the section to **Donors & Granters** with a bold title of **Donation** or **Grant  Award** for each respective party.

Also: Please add Aragon Nest since we were accepted into their program

julientregoat replied:
On it! Just gotta do this then PR should be good to go

prestonvanloon commented on src/assets/donors/index.js:
```diff
@@ -0,0 +1,33 @@
+    logo: EthereumLogo,
+    name: "Ethereum Foundation",
+    website: "https://www.ethereum.org/foundation",
+    twitter: "https://twitter.com/ethereum"
```
Is this property used somewhere?

julientregoat replied:
I was considering it early on but didn’t end up doing it. Will remove

prestonvanloon commented:
If this is ready for review again, can you update the screenshot please?

julientregoat commented:
Just updated!

Couple of code changes: I renamed `Donors` to `Contributors` so it would apply to both Granters and Donors. I also didn't update the links to point to the transaction hashes since I wasn't sure which transaction was from where.

prestonvanloon commented:
@rauljordan can you be more specific about more / less space?

rauljordan commented:
What I mean is there is a mismatch between the two that's very apparent to the viewer in terms of margins. It's just a small thing, but if we can at least make the margins between sections equal it would  do a lot for the home page.

prestonvanloon commented:
LGTM to me. @rauljordan?

terenc3t commented on src/about/donor_grants/DonorGrantsContainer.js:
```diff
@@ -0,0 +1,25 @@
+        work on sharding, furthering our goal to scale the Ethereum blockchain for all. We have set 
+        up a multisig smart contract to accept donations.</p>
+        <a href="https://etherscan.io/address/0x9B984D5a03980D8dc0a24506c968465424c81DbE">
+          Address: 0x9B984D5a03980D8dc0a24506c968465424c81DbE
```
can we add ens here too? it's prysmatic.eth

prestonvanloon replied:
Can we punt that to another pull request? 
The objective here is to add donors and it's been outstanding for almost 3 weeks. 

@julientregoat if you would like to fix this now then go ahead, otherwise I'd like to merge these changes as soon as we can.

rauljordan replied:
Let's do that in a different PR.

terenc3t commented:
That's fine

On Sat, May 26, 2018 at 8:58 AM Preston Van Loon <notifications@github.com>
wrote:

> *@prestonvanloon* commented on this pull request.
> ------------------------------
>
> In src/about/donor_grants/DonorGrantsContainer.js
> <https://github.com/prysmaticlabs/website/pull/16#discussion_r191052522>:
>
> > @@ -0,0 +1,25 @@
> +import React from 'react';
> +import './DonorGrantsContainer.css'
> +import Contributors from './Contributors'
> +
> +class DonorGrantsContainer extends React.Component {
> +
> +  render(){
> +    return (
> +      <div className="container donor-grants-container">
> +        <h2>Donors & Granters</h2>
> +        <p>Your support means a lot to us. With your help, we're able to accelerate our
> +        work on sharding, furthering our goal to scale the Ethereum blockchain for all. We have set
> +        up a multisig smart contract to accept donations.</p>
> +        <a href="https://etherscan.io/address/0x9B984D5a03980D8dc0a24506c968465424c81DbE">
> +          Address: 0x9B984D5a03980D8dc0a24506c968465424c81DbE
>
> Can we punt that to another pull request?
> The objective here is to add donors and it's been outstanding for almost 3
> weeks.
>
> @julientregoat <https://github.com/julientregoat> if you would like to
> fix this now then go ahead, otherwise I'd like to merge these changes as
> soon as we can.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/prysmaticlabs/website/pull/16#discussion_r191052522>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AUVDuVUj5XwVGYRH9TUJ6x7iURBl3fsjks5t2XuPgaJpZM4T0cMR>
> .
>


-- 
Terence Tsao

rauljordan commented:
Approved, thank you for this @julientregoat !
